solr/contrib/ltr/src/java/org/apache/solr/ltr/model/LTRScoringModel.java

```diff
@@ -158,7 +191,7 @@ public boolean hasParams() {
   }
 
   public Collection<Feature> getAllFeatures() {
-    return allFeatures;
+    return Collections.unmodifiableList(allFeatures);
```

Comment:
I like the unmodifiableList, but could it have impact on performance? are we creating a new (or several) object in the heap?